src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs'
+import { Kafka } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const kafkaOptions = await this.resolveProtocolConfig('kafka')
+    const securityRequirements = (this.AsyncAPIServer.security() || []).map(
+      (sec) => {
+        const secName = Object.keys(sec.json())[0]
+        return this.parsedAsyncAPI.components().securityScheme(secName)
+      }
+    )
+    const userAndPasswordSecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'userPassword'
+    )
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    this.kafka = new Kafka({
+      clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
+      brokers: [brokerUrl.host],
+      ssl: {
+        rejectUnauthorized: true,
+        key: fs.readFileSync('client.key'),
+        cert: undefined
+      },
+      sasl: {
+        mechanism: 'scram-sha-512',
```

Comment:
Alright so here is where I meant you should be guessing the mechanism. On line 21 above you do:

```js
const userAndPasswordSecurityReq = securityRequirements.find(
  (sec) => sec.type() === 'userPassword'
)
```

This is the same as saying: "please find a security requirement in the AsyncAPI file that has the type `userPassword`". If `userAndPasswordSecurityReq` is different than `undefined`, it means it found it. In such a case, `mechanism` in this line should be `plain`.

Now, you're missing other checks there. You also have to find if there are any security requirements of the type `scramSha256` or `scramSha512`, so you have to find them the same way:

```js
const scramSha256SecurityReq = securityRequirements.find(
  (sec) => sec.type() === 'scramSha256'
)
const scramSha512SecurityReq = securityRequirements.find(
  (sec) => sec.type() === 'scramSha512'
)
```

If `scramSha256SecurityReq` is different than `undefined` then `mechanism` should be `scramSha256`. If `scramSha512SecurityReq` is different than `undefined` then `mechanism` should be `scramSha512`. If all the 3 are `undefined` or the `securityRequirements` variable (on line 15) is `undefined` then `mechanism` should be `plain`.

I left an example above about how you can add a scramSha256 security requirement in your AsyncAPI file.

Reply:
ok, got it, so its basically one security type from user password, scramsha256, scramsha512 and update it in YAML file

Reply:
I've added a suggestion that you didn't apply yet to make the security mechanism of our example of type `scramSha256`. And yeah, if no mechanism is defined, it should be `plain` too.